pkgs/development/python-modules/bitsandbytes/default.nix

```diff
@@ -48,47 +53,55 @@ buildPythonPackage {
   inherit pname version;
   pyproject = true;
 
-  disabled = pythonOlder "3.7";
-
   src = fetchFromGitHub {
     owner = "TimDettmers";
     repo = "bitsandbytes";
     rev = "refs/tags/${version}";
-    hash = "sha256-GFbFKPdV96DXPA+PZO4h0zdBclN670fb0PGv4QPHWHU=";
+    hash = "sha256-yvxD5ymMK5p4Xg7Csx/90mPV3yxUC6QUuF/8BKO2p0k=";
   };
 
-  postPatch =
-    ''
-      substituteInPlace Makefile --replace "/usr/bin/g++" "g++" --replace "lib64" "lib"
-      substituteInPlace bitsandbytes/cuda_setup/main.py  \
-        --replace "binary_path = package_dir / self.binary_name"  \
-                  "binary_path = Path('$out/${python.sitePackages}/${pname}')/self.binary_name"
-    ''
-    + lib.optionalString torch.cudaSupport ''
-      substituteInPlace bitsandbytes/cuda_setup/main.py  \
-        --replace "/usr/local/cuda/lib64" "${cuda-native-redist}/lib"
-    '';
+  # By default, which library is loaded depends on the result of `torch.cuda.is_available()`.
+  # When `cudaSupport` is enabled, bypass this check and load the cuda library unconditionnally.
+  # Indeed, in this case, only `libbitsandbytes_cuda124.so` is built. `libbitsandbytes_cpu.so` is not.
+  # Also, hardcode the path to the previously built library instead of relying on
+  # `get_cuda_bnb_library_path(cuda_specs)` which relies on `torch.cuda` too.
+  #
```

Comment:
```suggestion
```

Reply:
It was on purpose to make this long comment more readable. I can remove it if necessary.

Reply:
oh oops, github's diff viewer showed there being an empty line between this and line 69. i misunderstood -- sorry for the noise!